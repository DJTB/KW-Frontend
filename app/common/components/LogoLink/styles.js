import { Link } from 'react-router-dom';
import styled, { css } from 'styled-components';
import { timingFunctions } from 'polished';

import Logo from '-!babel-loader!svg-react-loader!common/assets/img/logo.svg';
import { blueLight, purpleLight, pink } from 'common/styles/colors';

const linkStyle = css`
  display: block;
  position: relative;
  vertical-align: middle;
  align-self: center;
  flex: 0 0 auto;
  width: ${({ size }) => size}; /*CSS instead of html width attr to support non-pixel units*/
  height: ${({ size }) => size}; /*Prevents scaling issue in IE*/
  background-repeat: no-repeat;
  color: ${blueLight};
  cursor: pointer;
  transition: color .4s ${timingFunctions('easeInOutSine')};

  &:hover {
    transition: color .5s ${timingFunctions('easeOutQuad')};
    color: ${purpleLight};
  }

  &:active {
    transition: color .1s ${timingFunctions('easeOutQuad')};
    color: ${pink};
  }
`;

const logoStyle = css`
  display: block;
  position: absolute;
  transform-origin: 50% 50% 0px;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;

  .bg {
    color: inherit;
    fill: currentColor;
  }
`;


export const RouterLink = styled(Link)`${linkStyle}`;
export const Svg = styled(Logo)`${logoStyle}`;
