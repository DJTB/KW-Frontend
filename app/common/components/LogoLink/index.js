import React from 'react';
import PropTypes from 'prop-types';

import { debounce } from 'lodash';
import { breakpoints } from 'common/styles/media';
import { RouterLink, Svg } from './styles';

export const LOGO_LARGE_REMS = '4.25rem';
export const LOGO_SMALL_REMS = '3.25rem';

const getSize = () => (window.innerWidth > breakpoints.md ? LOGO_LARGE_REMS : LOGO_SMALL_REMS);

class LogoLink extends React.Component {
  static propTypes = {
    to: PropTypes.string,
  };

  static defaultProps = {
    to: '/',
  };

  state = {
    size: getSize(),
  };

  componentDidMount() {
    window.addEventListener('resize', this.handleResize);
  }

  componentWillUnmount() {
    window.removeEventListener('resize', this.handleResize);
  }

  handleResize = debounce(() => {
    this.setState({ size: getSize() });
  }, 300);

  render() {
    return (
      <RouterLink to={this.props.to} size={this.state.size} title="Return Home">
        <Svg width="100%" height="100%" />
      </RouterLink>
    );
  }
}

export default LogoLink;
