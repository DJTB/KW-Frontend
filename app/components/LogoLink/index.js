import React from 'react';
import PropTypes from 'prop-types';

import { Link, Svg } from './styles';

LogoLink.propTypes = {
  to: PropTypes.string,
  size: PropTypes.string,
};

LogoLink.defaultProps = {
  to: '/',
  size: '4.5rem',
};

function LogoLink({ to, size }) {
  return (
    <Link to={to} size={size} title="Return Home">
      <Svg width="100%" height="100%" />
    </Link>
  );
}

export default LogoLink;
