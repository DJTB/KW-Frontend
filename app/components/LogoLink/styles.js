import { Link as RouterLink } from 'react-router';
import styled, { css } from 'styled-components';
import { timingFunctions } from 'polished';

import Logo from '-!babel-loader!svg-react-loader!shared/assets/img/logo.svg';
import { blueLight, purpleLight, pink } from 'shared/styles/colors';

const linkStyle = css`
  display: block;
  vertical-align: middle;
  align-self: center;
  position: relative;
  width: ${({ size }) => size}; /*CSS instead of html width attr to support non-pixel units*/
  height: ${({ size }) => size}; /*Prevents scaling issue in IE*/
  background-repeat: no-repeat;
  color: ${blueLight};
  cursor: pointer;
  transition: color .4s ${timingFunctions('easeInOutSine')};

  &:hover {
    transition: color .5s ${timingFunctions('easeOutQuad')};
    color: ${purpleLight};
  }

  &:active {
    transition: color .1s ${timingFunctions('easeOutQuad')};
    color: ${pink};
  }
`;

const logoStyle = css`
  display: block;
  position: absolute;
  transform-origin: 50% 50% 0px;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;

  .bg {
    color: inherit;
    fill: currentColor;
  }
`;


export const Link = styled(RouterLink)`${linkStyle}`;
export const Svg = styled(Logo)`${logoStyle}`;
