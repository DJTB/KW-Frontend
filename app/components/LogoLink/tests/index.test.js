import 'jest-styled-components';
import React from 'react';
import { shallow } from 'enzyme';
import LogoLink from '../index';

describe('<LogoLink />', () => {
  it('should match baseline snapshot', () => {
    expect(shallow(<LogoLink />)).toMatchSnapshot();
  });
});
