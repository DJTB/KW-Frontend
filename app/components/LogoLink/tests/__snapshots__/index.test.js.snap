// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LogoLink /> should match baseline snapshot 1`] = `
<Styled(Link)
  size="4.5rem"
  title="Return Home"
  to="/"
>
  <styled.IMAGE_MOCK
    height="100%"
    width="100%"
  />
</Styled(Link)>
`;
